app.rb

```diff
@@ -82,13 +82,18 @@ def user
       org.apache.shiro.SecurityUtils.subject
     end
 
+    # Check if request is from localhost and if bypass for localhost is enabled
+    def local_request?
+      request.ip == '127.0.0.1' and Razor.config['auth.allow_localhost']
```

Comment:
The IP is coming from the ENV REMOTE_ADDR and is set by the webserver.
Therefore spoofing from an external system is not possible.

Depending on the setup HTTP_X_FORWARDED_FOR is considered as well.
(https://github.com/rack/rack/blob/master/lib/rack/request.rb#L273).

Rails is just using the REMOTE_ADDR header.
(http://api.rubyonrails.org/classes/ActionDispatch/Request.html#method-i-local-3F).